### PR TITLE
refactor! feat: Add aten::reflection_pad2d and aten::reflection_pad1 converter support.

### LINKED ISSUE #775: 🐛 [Bug] Create tensor with negative dimension when  compileTorch-TensorRT
##  Bug Description

I'm trying to compile but getting this error:

> GRAPH: [Torch-TensorRT] - Node we are looking at: %771 : Tensor = aten::add(%mel, %582, %583) # /root/Project/experiment/Torch-TensorRT/VocGAN/model/generator.py:315:0
GRAPH: [Torch-TensorRT] - Node %771 : Tensor = aten::add(%mel, %582, %583) # /root/Project/experiment/Torch-TensorRT/VocGAN/model/generator.py:315:0 outputs a tensor
GRAPH: [Torch-TensorRT] - Input to node: %mel : Tensor = prim::Param()
GRAPH: [Torch-TensorRT] - Input outputs a Tensor
GRAPH: [Torch-TensorRT] - Input to node: %582 : Tensor = prim::Constant[value={5}]()
GRAPH: [Torch-TensorRT] - Input outputs a Tensor
GRAPH: [Torch-TensorRT] - Input is a constant
GRAPH: [Torch-TensorRT] - Found that constant tensor has type: Double
GRAPH: [Torch-TensorRT] - Estimated input type is Double
DEBUG: [Torch-TensorRT] - Paring 0: mel : Input(shape: [1, 80, -1], min: [1, 80, 99], opt: [1, 80, 100], max: [1, 80, 101], dtype: Float32, format: NCHW\Contiguous\Linear)
WARNING: [Torch-TensorRT] - For input mel, found user specified input dtype as Float32, however when inspecting the graph, the input type expected was inferred to be Double
The compiler is going to use the user setting Float32
This conflict may cause an error at runtime due to partial compilation being enabled and therefore
compatibility with PyTorch's data type convention is required.
If you do indeed see errors at runtime either:
Remove the dtype spec for mel
Disable partial compilation by setting require_full_compilation to True
 Traceback (most recent call last):
  File "torchscript_2_tsRT.py", line 76, in <module>
    trt_ts_module = torch_tensorrt.ts.compile(vocgan_trace, **compile_settings)
  File "/opt/conda/lib/python3.8/site-packages/torch_tensorrt/ts/_compiler.py", line 119, in compile
    compiled_cpp_mod = _C.compile_graph(module._c, _parse_compile_spec(spec))
RuntimeError: Trying to create tensor with negative dimension -1: [1, 80, -1]

## To Reproduce

https://github.com/rishikksh20/VocGAN
```
import sys
sys.path.insert(1, '/root/Project/experiment/Torch-TensorRT/VocGAN')
import torch_tensorrt as torchtrt
import torch_tensorrt
from torch_tensorrt import _enums
import torch
import numpy as np
from utils.hparams import HParam, load_hparam_str
from model.generator import ModifiedGenerator
from scipy.io.wavfile import write


def vocoder_post_process_2(float_wav):
    hop_length = 256
    MAX_WAV_VALUE = 32768.0

    audio = float_wav[:-(hop_length*10)]
    audio = MAX_WAV_VALUE * audio
    audio = audio.clamp(min=-MAX_WAV_VALUE, max=MAX_WAV_VALUE-1)
    audio = audio.short()
    audio = audio.cpu().detach().numpy()

    write('torchscript.wav', 22050, audio)


new_level = torch_tensorrt.logging.Level.Graph
torch_tensorrt.logging.set_reportable_log_level(new_level)

checkpoint = torch.load("VocGAN/vocoder.pt")

hp = load_hparam_str(checkpoint['hp_str'])

model = ModifiedGenerator(hp.audio.n_mel_channels, hp.model.n_residual_layers,
                    ratios=hp.model.generator_ratio, mult = hp.model.mult,
                    out_band = hp.model.out_channels).cuda()
model.load_state_dict(checkpoint['model_g'])
model.eval(inference=True)
inputs_test = torch.ones(1, 80).to('cuda').float()
print(a.size())
with torch.no_grad():
    mel = torch.from_numpy(np.load("VocGAN/mel.npy"))
    #print(mel.type())
    if len(mel.shape) == 2:
        mel = mel.unsqueeze(0)
    mel = mel.cuda()
    zero = torch.full((1, 80, 10), -11.5129).to(mel.device)
    mel = torch.cat((mel, zero), dim=2)
    vocgan_trace = torch.jit.trace(model, mel)

compile_settings = {
    "inputs": [torch_tensorrt.Input(
        min_shape=[1, 80, 10],
        opt_shape=[1, 80, 100],
        max_shape=[1, 80, 1000],
        dtype=torch.float,
        #format=torch_tensorrt.TensorFormat.NCHW
        #format=_enums.TensorFormat.contiguous 
    )],
    #"debug": True,
    "enabled_precisions": {torch.float},  # Run with FP16
    #"max_batch_size":3,
    "truncate_long_and_double":True,
    "require_full_compilation":False,
    "torch_executed_ops":[
            "aten::reflection_pad1d"
    ]
}

trt_ts_module = torch_tensorrt.ts.compile(vocgan_trace, **compile_settings)
audio = trt_ts_module(mel)
audio = audio.squeeze(0) 
wav_pred = vocoder_post_process_2(audio.squeeze())

#torch_tensorrt_out = trt_ts_module(mel)
torch.jit.save(trt_ts_module, 'tsRT.jit')
```

## Expected behavior

The model is compiled to TensorRT without error.

## Environment
 - Torch-TensorRT Version (e.g. 1.0.0): v1.0.0
 - PyTorch Version (e.g. 1.0): 1.0
 - CPU Architecture: x86 intel
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): docker pull nvcr.io/nvidia/pytorch:21.11-py3
 - CUDA version:  NVIDIA-SMI 495.29.05    Driver Version: 495.29.05    CUDA Version: 11.5
 - GPU models and configuration: 1050ti

## Additional context
 If I change min, opt, max_shape into the same size like [1. 80. 200], compiling will be good but inference result is not.
If the last dimension of min, opt, max_shape not the same, I got "ntimeError: Trying to create tensor with negative dimension -1: [1, 80, -1]"

### PR DESCRIPTION
**Pull request by p1x31**
Signed-off-by: Vadim Makarenkov <vadim_makarenkov@hotmail.co.uk>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Refactor of #611 for the 1.1.0a0+b652045d version
Add aten::reflection_pad2d and aten::reflection_pad1d support.
All credits to @chengyute 

Fixes #775 (full compilation) #545 

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# Checklist:

- [X] My code follows the style guidelines of this project (You can use the linters)
- [X] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [X] I have added tests to verify my fix or my feature
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by p1x31** [comment]
These converters only supported in [TensorRT 8.2.0.6 EA](https://docs.nvidia.com/deeplearning/tensorrt/release-notes/tensorrt-8.html#rel-8-2-0-EA) release

**Comment by narendasan** [comment]
We are about to land 8.2 in master, will merge this after

**Review by peri044 — APPROVED** [review]
Please fix the linter issue. The code LGTM

File: tests/core/conversion/converters/test_reflection_pad.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        %5 : int[] = prim::ListConstruct(%1, %2, %3, %4)
+        %6 : Tensor = aten::reflection_pad2d(%0, %5)
+        return (%6))IR";
+
+  auto g = std::make_shared<torch::jit::Graph>();
+  torch::jit::parseIR(graph, g.get());
+
+  auto in1 = at::randint(1, 10, {1, 3, 5, 5}, {at::kCUDA});
+
+  auto params = torch_tensorrt::core::conversion::get_named_params(g->inputs(), {});
```
Comment (peri044) [inline tests/core/conversion/converters/test_reflection_pad.cpp:24]: `get_named_params` is now replaced with `get_static_params`. Please change this and other similar lines with 
` params = torch_tensorrt::core::ir::get_static_params(g->inputs(), {});`

**Review by peri044 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**